### PR TITLE
DB scripts

### PR DESCRIPTION
**Pull request by glpatcern**
@javfg and @gmgigi96 this is all I learned so far about the CERNBox/OC database

### CODE REVIEW

File: db/setup_notifications.sql [hunk 1]
```diff
@@ -45,9 +45,5 @@ CREATE INDEX `notification_recipients_ix1` ON `notification_recipients` (`recipi
 
 -- changes for added notifications on oc shares
 
-ALTER TABLE oc_share ADD notify_uploads BOOL DEFAULT false;
+ALTER TABLE oc_share ADD notify_uploads INTEGER DEFAULT 0 NOT NULL;
```
Comment (javfg) [inline db/setup_notifications.sql]: Keep in mind this would require changes in the web and reva too!

Comment (glpatcern) [reply db/setup_notifications.sql]: That's why I said in #4061 that we need to test. The code in Reva does not seem to depend on the type, that is for now we can use the integer as a bool. Not sure about web.

Comment (javfg) [reply db/setup_notifications.sql]: I'm not 100% sure, I remember doing `strconv.ParseBool` in some places but not sure about all of them. I guess as long as we only store 0 or 1 and we can parse it, it should be OK.

Comment (glpatcern) [reply db/setup_notifications.sql]: Anyway given the change of table name, the whole thing is to be retested after merging cs3org#4061 (which you should review ;) ).

File: db/drop_oc_unused_tables.sql [hunk 2]
```diff
@@ -0,0 +1,48 @@
+DROP TABLE IF EXISTS `oc_account_terms`;
```
Comment (gmgigi96) [inline db/drop_oc_unused_tables.sql]: Why this has to go in the reva codebase? It's just CERNBox related

File: db/drop_old_cernbox_tables.sql [hunk 3]
```diff
@@ -0,0 +1,2 @@
+DROP TABLE IF EXISTS `cbox_canary`;
```
Comment (gmgigi96) [inline db/drop_old_cernbox_tables.sql]: Same as above

File: db/init.sql [hunk 4]
```diff
@@ -0,0 +1,398 @@
...
+
+--
+-- Table structure for table `oc_sciencemesh`
+--
+
+DROP TABLE IF EXISTS `oc_sciencemesh`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `oc_sciencemesh` (
+  `iopurl` varchar(255) COLLATE utf8_bin NOT NULL DEFAULT 'http://localhost:10999',
```
Comment (gmgigi96) [inline db/init.sql:127]: Why this default??

File: db/init.sql [hunk 5]
```diff
@@ -0,0 +1,398 @@
...
+  `country` varchar(2) COLLATE utf8_bin NOT NULL,
+  `hostname` varchar(255) COLLATE utf8_bin NOT NULL DEFAULT 'example.org',
+  `sitename` varchar(255) COLLATE utf8_bin NOT NULL DEFAULT 'CERNBox',
```
Comment (gmgigi96) [inline db/init.sql:130]: Why CERNBox as default?

File: db/init.sql [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+  `siteurl` varchar(255) COLLATE utf8_bin NOT NULL DEFAULT 'http://localhost',
```
Comment (gmgigi96) [inline db/init.sql:131]: Why this default?

File: db/init.sql [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+-- Table structure for table `oc_sciencemesh`
+--
+
+DROP TABLE IF EXISTS `oc_sciencemesh`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `oc_sciencemesh` (
+  `iopurl` varchar(255) COLLATE utf8_bin NOT NULL DEFAULT 'http://localhost:10999',
+  `country` varchar(2) COLLATE utf8_bin NOT NULL,
+  `hostname` varchar(255) COLLATE utf8_bin NOT NULL DEFAULT 'example.org',
```
Comment (gmgigi96) [inline db/init.sql:129]: Why this default?

File: db/init.sql [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+/*!40101 SET character_set_client = @saved_cs_client */;
+
+--
+-- Table structure for table `oc_share`
+--
+
+DROP TABLE IF EXISTS `oc_share`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!40101 SET character_set_client = utf8 */;
+CREATE TABLE `oc_share` (
```
Comment (gmgigi96) [inline db/init.sql:145]: Maybe we could review this table and normalize it

**Comment by gmgigi96** [comment]
@glpatcern yesterday I didn’t notice this was the cernbox fork. Even though this fork will be definitely merged with upstream soon, the schema is ok for me

**Comment by glpatcern** [comment]
In fact, @gmgigi96 your questions are very relevant. I propose the following:
* Anything to do with https://github.com/cernbox/revaold or the legacy OC database should stay somewhere in the CERN gitlab
* In this repo we should keep an up-to-date `init.sql` and the update scripts "as they come"

**Comment by glpatcern** [comment]
For the record, the legacy scripts are now in https://gitlab.cern.ch/cernbox/cbox-scripts/-/tree/master/owncloud-db, so what remains here is entirely for Reva (with all the caveats observed by @gmgigi96).

**Comment by gmgigi96** [comment]
@glpatcern with all the new changes, LGTM!

**Pull request merged**